libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -45,6 +47,109 @@ static int lua_micros(lua_State *L) {
     return 1;
 }
 
+static int lua_mavlink_receive(lua_State *L) {
+    check_arguments(L, 0, "receive");
+
+    struct AP_Scripting::mavlink_msg msg;
+    ObjectBuffer<struct AP_Scripting::mavlink_msg> *input = AP::scripting()->mavlink_data.input;
+
+    if (input == nullptr) {
+        luaL_error(L, "Never subscribed to a message");
+    }
+
+    if (input->pop(msg)) {
```

Comment:
Don't we need the semaphore to pop?

Reply:
Not if you only read from one thread. The ring buffers are thread safe as long as their is a single publisher, and a single consumer. (The nullptr check is the only to really worry about)

Reply:
right, as long as you stick to push/pop. If for example you do a forced push then it becomes unsafe (as it is effectively a pop and push in one) or if you resize.